src/CachePoolTest.php

```diff
@@ -520,7 +540,11 @@ public function testGetItemsInvalidKeys($key)
             return;
         }
 
-        $this->cache->getItems(['key1', $key, 'key2']);
+        $items = $this->cache->getItems(['key1', $key, 'key2']);
+
+        if ($items instanceof \Traversable) {
+            iterator_to_array($items);
```

Comment:
Well, this is an issue in your implementation then. The PSR spec says that the exception must be thrown by ``getItems``, not by the iteration over items. This is absolutely not equivalent (try to catch the exception and you will see it does not work the same).
And this is precisely why I'm rejecting your change here: it makes the test accept invalid implementations